src/server/util/parse.ts

```diff
@@ -0,0 +1,29 @@
+import { emailValidator, ogHostname } from '../config'
+
+/**
+ * Remove wildcard and escape characters from the query.
+ * @param query The query to sanitise.
+ */
+export function sanitiseQuery(query: string) {
+  // check legit domain
+  if (emailValidator.match(query)) {
+    // remove wildcards characters and escape characters
+    const inputRaw = query.replace(/(%|\\)/g, '')
+    return `%${inputRaw}`
+  }
+
+  return ''
+}
+
+/**
+ * Checks for full short link URL in the query and extract out the short link.
+ * @param query The query to check.
+ */
+export function extractShortUrl(query: string) {
+  const regExp = new RegExp(`^https://${ogHostname}/(.*)`)
+  const capture = regExp.exec(query)
+  if (capture) {
+    return capture[1]
+  }
+  return query
```

Comment:
returning `query` makes this function a misnomer. this function extracts the shortURL. if there's no shortURL to be found, my impression is that `nothing` would be extracted. would returning `null` be a more valid behaviour here?

then other functions that call it would check if there are any shortURLs that can be extracted, if not, use whatever fallback mechanisms available. either way, it's not the job of this function to fallback for them.